argo-workflows/okd-workflows/30-build-workflow-template.yaml

```diff
@@ -0,0 +1,185 @@
+          EOF
+          
+          image=$(/tmp/yq -r '.spec.output.to.name' /tmp/build.yaml)
+          namespace=$(/tmp/yq -r '.spec.output.to.namespace' /tmp/build.yaml)
+          if [ "$namespace" == "null" ]; then
+            # Use the namespace associated with the service account running this container
+            namespace=$(</var/run/secrets/kubernetes.io/serviceaccount/namespace)
+          fi
+          echo "image-registry.openshift-image-registry.svc:5000/$namespace/${image%-*}" > /tmp/final-image
+    - name: build-run-and-wait
+      inputs:
+        parameters:
+          - name: build
+      resource:
+        action: create
+        setOwnerReference: true
+        successCondition: status.phase in (Complete)
+        failureCondition: status.phase in (Failed)
+        manifest: |
+          {{inputs.parameters.build}}
+    - name: compose-manifest-list
+      inputs:
+        parameters:
+          - name: final-image
+          - name: architectures
+      script:
+        image: image-registry.openshift-image-registry.svc:5000/openshift/cli
+        command:
+          - bash
+        source: |
+          #!/bin/bash
+          set -euo pipefail
+          # TODO another image? The current official manifest-tool image extends from scratch 
+          # and we cannot use files redirection in the args list if using it
+          wget -O /tmp/binaries-manifest-tool.tar.gz https://github.com/estesp/manifest-tool/releases/download/v2.0.8/binaries-manifest-tool-2.0.8.tar.gz
+          tar -C /tmp --transform 's/manifest-tool-linux-.*/mtl/' \
+            -xvf /tmp/binaries-manifest-tool.tar.gz manifest-tool-linux-$(uname -m | sed 's/aarch64/arm64/;s/x86_64/amd64/')
+          image='{{inputs.parameters.final-image}}'
+          image="${image%\",*}"
+          image="${image#\[\"}"
+          mapfile -t architectures < <(echo "{{inputs.parameters.architectures}}" | tr ',' '\n')
+          echo "[INFO] Creating manifest list ${image} with manifests for architectures ${architectures[*]}"         
+          MANIFEST_LIST="
+          image: ${image}
+          manifests:"
+          for arch in "${architectures[@]}"; do
+            MANIFEST_LIST="${MANIFEST_LIST}
+            - image: ${image}-${arch}
+              platform:
+                architecture: ${arch}
+                os: linux"
+          done
+          echo "${MANIFEST_LIST}"          
+          /tmp/mtl --insecure --username k8s --password "$(</var/run/secrets/kubernetes.io/serviceaccount/token)" \
+            push from-spec <(echo "${MANIFEST_LIST}")
+          
+          echo "[INFO] Manifest list pushed to ${image}. Deleting individual manifests..."
+          set -x
+          image="${image#image-registry.openshift-image-registry.svc:5000/}"
+          namespace="${image%%/*}"
+          image="${image#*/}"
+          for arch in "${architectures[@]}"; do
+            oc delete -n ${namespace} istag "${image}-${arch}"
```

Comment:
this only deletes the tags..does the image get gc'ed?

Reply:
Does it? I did only short time tests till now and have to verify it yet (good point)

Reply:
they are istags pushed from the Builds v1